### ISSUE TITLE
recurring fatal: It seems that there is already a rebase-merge directory errors

### ISSUE THREAD

**Issue by spicemix** [Homebrew/brew#11142]
### `brew config` output

```shell
HOMEBREW_VERSION: 3.1.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: bc9b98aa0ba777889218ac51f67ccc76b0be6a28
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 685eea39ffe258bb79e323d212c2e1ff020eefc4
Core tap last commit: 78 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0 build 1200
Git: 2.31.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: N/A
Xcode: 12.4
```

### `brew doctor` output

```shell
Your system is ready to brew.
```

### 

I ran `brew update` and am still able to reproduce my issue.
I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

brew update; brew upgrade; brew cleanup

### What happened (include all command output)?
```shell
fatal: It seems that there is already a rebase-merge directory, and
I wonder if you are in the middle of another rebase.  If that is the
case, please try
	git rebase (--continue | --abort | --skip)
If that is not the case, please
	rm -fr ".git/rebase-merge"
and run me again.  I am stopping in case you still have something
valuable there.

Updated Homebrew from 586d25a52 to bc9b98aa0.
Updated 2 taps (homebrew/cask-versions and homebrew/cask).
```
### What did you expect to happen?

Normal functionality.

When I see this error, I run `brew update-reset` as was recommended here earlier. After that, I do get normal functionality for a few weeks, but then this error comes back. It may be showing up on homebrew updates.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew update; brew upgrade; brew cleanup
```

**Comment by MikeMcQuaid** [comment]
> When I see this error, I run `brew update-reset` as was recommended here earlier. After that, I do get normal functionality for a few weeks, but then this error comes back. It may be showing up on homebrew updates.

Are you deliberately modifying any of Homebrew's files?

**Comment by spicemix** [comment]
No I have never to my knowledge done anything weird. Is there an easy full reinstall procedure? I have used homebrew for many years and always kept current.

**Comment by carlocab** [comment]
Use `brew bundle bunde` to create a `Brewfile` in your working directory for easy reinstallation. Then completely reinstall Homebrew. You can start with the uninstall script, and you'll probably need to do a little extra cleanup after that.

**Comment by spicemix** [comment]
Thanks Carlo although that was somewhat illegible. What I did was use instructions on [brew bundle](https://gist.github.com/ChristopherA/a579274536aab36ea9966f301ff14f3f) and [installing homebrew](https://github.com/homebrew/install#uninstall-homebrew). In the terminal in my ~ user directory:

`brew bundle dump`

Here I probably should have done:

`brew bundle cleanup`

...but I didn't and got a bunch of install errors (app already exists) later. But maybe this way I saved my preferences?

Then:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
`

`open /usr/local`

(Manually delete a bunch of brew-related stuff.)

Then back in the terminal in the same directory I dumped the Brewfile:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
`

`brew bundle install`

I can mark this closed for now and hopefully be able to reopen it if this full reinstall didn't remove the issue. I am pretty sure it was only showing up on homebrew updates (and on all of them). Probably I didn't transition between homebrew versions cleanly when things got refactored a while back. Thanks!
